File: ios/MobileToolkit/Sources/MobileToolkit/MobileToolkitPlugin.swift
import Foundation

@objc public class MobileToolkitPlugin: NSObject {
    private var features: [String: Feature] = [:]
    
    public override init() {
        super.init()
    }
    
    internal func registerFeature(name: String, feature: Feature) {
        features[name] = feature
    }
    
    @objc public func initialize(_ config: [String: Any]) {
        features.values.forEach { $0.initialize(config) }
    }
}

File: android/mobiletoolkit/src/main/kotlin/com/carnaxidegames/mobiletoolkit/MobileToolkitPlugin.kt
package com.carnaxidegames.mobiletoolkit

import org.godotengine.godot.Godot
import org.godotengine.godot.plugin.GodotPlugin
import org.godotengine.godot.plugin.UsedByGodot

class MobileToolkitPlugin(godot: Godot) : GodotPlugin(godot) {
    override fun getPluginName() = "MobileToolkit"
    
    private val features = mutableMapOf<String, Feature>()
    
    override fun getPluginSignals(): Set<SignalInfo> {
        return features.values.flatMap { it.getSignals() }.toSet()
    }
    
    internal fun registerFeature(name: String, feature: Feature) {
        features[name] = feature
    }
    
    @UsedByGodot
    fun initialize(config: Dictionary) {
        features.values.forEach { it.initialize(config) }
    }
}